### PR TITLE
Bug 1159190 - Add a commit hook which forbids mozL10n.get

### PR DESCRIPTION
**Pull request by stasm**
https://bugzilla.mozilla.org/show_bug.cgi?id=1159190

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=66b0e186c72e892b122a72257ece2c04fb7f2505

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0df22d6d3418dbb7e5826860f0cf86c29f623f1b

File: tools/pre-commit [hunk 1]
```diff
@@ -9,11 +9,16 @@ JSHINT_EXECUTABLE="node_modules/.bin/jshint"
 JSHINT_XFAIL_LIST="build/jshint/xfail.list"
 JSHINT_ARGS="--reporter=build/jshint/xfail $JSHINT_ARGS"
 excluded_list=".jshintignore"
+L10N_XFAIL_LIST="build/l10n/xfail.list"
```
Comment (julienw) [inline tools/pre-commit]: should we create a file with existing failing files?

File: tools/pre-commit [hunk 2]
```diff
@@ -86,6 +91,34 @@ then
...
+      fi
+    done
+
+    test -n "$one_was_ignored" && mozL10n_information
+
+    to_lint=`echo "$js_changed_files" | sort | comm -13 "$L10N_XFAIL_LIST" - | tr '[:space:]' ' '`
+    # check for deprecated mozL10n API uses
+    for file in $to_lint ; do
+      one_failed=""
+      if grep -q "mozL10n.get" "$file" ; then
```
Comment (julienw) [inline tools/pre-commit]: I think you want `-F`

Comment (julienw) [inline tools/pre-commit]: This is easily escapable in the code too; maybe file a separate bug to make it work in eslint and add the bug number in a comment here ?

File: tools/pre-commit [hunk 3]
```diff
@@ -86,6 +91,34 @@ then
...
+        one_failed="1"
+        echo "$file uses mozL10n.get which is deprecated"
```
Comment (julienw) [inline tools/pre-commit]: nit: add a full stop at the end of the sentence

File: tools/pre-commit [hunk 4]
```diff
@@ -86,6 +91,34 @@ then
     test $jshint_result -ne 0 && jshint_information
     exit 1
   fi
+
```
Comment (julienw) [inline tools/pre-commit]: nit: add an empty echo to better separate the checks' outputs

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4858c6f9972d94a3f9f0f1e29bda896469e0b4f6

File: tools/pre-commit (hunk 1 above)
Comment (stasm) [inline tools/pre-commit]: At first I thought we'd start small, with an empty file, and grow it as needed, but I ended up adding the full list.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7f4a9dfc494d9dbb8d5d3f718401e58142ca15a5

**Pull request merged**